c7n/policy.py

```diff
@@ -72,9 +74,20 @@ def __init__(self, data, options):
 
         # We store all the policies passed in so we can refilter later
         self._all_policies = []
+        session = utils.get_profile_session(options)
         for p in self.data.get('policies', []):
-            self._all_policies.append(
-                Policy(p, options, session_factory=self.test_session_factory()))
+            all_regions = session.get_available_regions(p['resource'])
+            if 'all' in options.regions:
+                options.regions = all_regions
+            for region in options.regions:
+                if region not in all_regions:
+                    # TODO - do we want a message
+                    continue
+                options_copy = copy.copy(options)
```

Comment:
rather than using copy.copy machinery, using a c7n.handler config class here would allow for clean overiding on a new copy. i had to do a hand verification that this is correct on shallow copy, plus the copy machinery is expensive.